File: packages/server/accounts/src/adapters/providers/jwt.ts
import * as Provider from 'jsonwebtoken';
import { JwtProvider, JwtSignOptions, JwtSignPayload } from '@/ports/providers';

export class JwtProviderImpl implements JwtProvider {
  public sign(payload: JwtSignPayload, secret: string, options: JwtSignOptions): string {
    return Provider.sign(payload, secret, options);
  }

  /** @todo check if is working */
  public verify(token: string, secret: string): string {
    const sub = Provider.verify(token, secret) as any;

    if (!sub) {
      throw new Error('Sub value not found');
    }

    const { userId } = sub;

    if (!userId) {
      throw new Error('Sub value is invalid');
    }

    return String(userId);
  }
}

File: packages/server/invites/src/adapters/providers/jwt.py
import jwt
from ...ports.providers import JwtProvider


class JwtProviderImpl(JwtProvider):
  def __init__(self, secret: str) -> None:
    self._secret_ = secret

  def verify(self, token: str):
    return jwt.decode(token, self._secret_, algorithms=["HS256"])
